src/paginator-spec.js

```diff
@@ -0,0 +1,486 @@
+        const requestCallCount = paginator.first()
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(1);
+      });
+    });
+
+    describe('last', function() {
+      beforeEach(function() {
+        paginator = new Paginator(request.spy);
+      });
+
+      it('returns a promise that successfully resolves', function() {
+        const promise = paginator.last();
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('returns a promise with a response instance', function() {
+        const response = paginator.last();
+
+        return expect(response).to.eventually.be.an.instanceOf(Response);
+      });
+
+      it('sets the current page to the last page', function() {
+        expect(paginator.page).to.equal(1);
+
+        const currentPage = paginator.last().then(getCurrentPage);
+
+        return expect(currentPage).to.eventually.equal(10);
+      });
+
+      it('calls the request given to the paginator', function() {
+        const requestCallCount = paginator.last()
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(2);
+      });
+    });
+
+    describe('current', function() {
+      beforeEach(function() {
+        paginator = new Paginator(request.spy);
+      });
+
+      it('returns a promise that successfully resolves', function() {
+        const promise = paginator.current();
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('doesn\'t change the current page', function() {
+        expect(paginator.page).to.equal(1);
+
+        const currentPage = paginator.current().then(getCurrentPage);
+
+        return expect(currentPage).to.eventually.equal(1);
+      });
+
+      it('calls the request given to the paginator', function() {
+        const requestCallCount = paginator.last()
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(2);
+      });
+    });
+
+    describe('fetchPage', function() {
+      const getUniqueValue = function(response) {
+        return response.json().then((body) => body.__unique__);
+      };
+
+      beforeEach(function() {
+        paginator = new Paginator(request.spy);
+      });
+
+      it('returns a promise that successfully resolves', function() {
+        const promise = paginator.fetchPage(1);
+
+        return expect(promise).to.eventually.resolve;
+      });
+
+      it('returns a promise with a response instance', function() {
+        const response = paginator.fetchPage(1);
+
+        return expect(response).to.eventually.be.an.instanceOf(Response);
+      });
+
+      it('calls the request given to the paginator', function() {
+        const requestCallCount = paginator.fetchPage(1)
+          .then(request.getCallCount);
+
+        return expect(requestCallCount).to.eventually.equal(1);
+      });
+
+      it('clone responses', function() {
```

Comment:
Hmm. This seems unintuitive. Every other method shouldn't be making duplicate requests, instead caching them and returning that for duplicate calls.

For instance, if I'm on page 2 and call `next`, `next`, `prev`, `prev`, `next` I'd expect 2 requests to be made and both `prev` calls and the last `next` call to return caches. I think `fetchPage` should be the same.